docs/howto/operator/gcp/gcs_to_gcs.rst

```diff
@@ -0,0 +1,85 @@
+
+Cloud Storage Transfer Service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are many operators that manage the Google Cloud Data Transfer service. If you want to create a new data transfer
+task, use the operator  :class:`~airflow.gcp.operators.cloud_storage_transfer_service.GcpTransferServiceJobCreateOperator`
+You can also use the previous operator for this service -
+:class:`~airflow.gcp.operators.cloud_storage_transfer_service.GoogleCloudStorageToGoogleCloudStorageTransferOperator`
+
+These operators does not control the copying process locally, but uses Google resources, which allows to
+perform this task faster and more economically. The economic effects are especially prominent when the
+existence of Airflow is not found in the Google Cloud Platform, because this operator allows egress
+traffic reductions.
+
+This operator modifies source objects if the option that specifies whether objects should be deleted
+from the source after they are transferred to the sink is enabled.
+
+When you use this service, you can specify whether overwriting objects that already exist in the sink is
+allowed, whether objects that exist only in the sink should be deleted, or whether objects should be deleted
+from the source after they are transferred to the sink.
+
+Source objects can be specified using include and exclusion prefixes, as well as based on the file
+modification date.
+
+If you need information on how to use it, look at the guide: :doc:`transfer`
+
+Local transfer
+~~~~~~~~~~~~~~
+
+There are two operators that are used to copy data, where the entire process is controlled locally.
+
+GoogleCloudStorageToGoogleCloudStorageOperator
+----------------------------------------------
+
+:class:`~airflow.operators.gcs_to_gcs.GoogleCloudStorageToGoogleCloudStorageOperator` allows you to copy
+one or more files. The copying always takes place without taking into account the initial state of
+the destination bucket.
+
+This operator never deletes data in the destination bucket and it modifies objects in the source bucket
+if the file move option is active.
+
+When you use this operator, you can specify whether objects should be deleted from the source after
+they are transferred to the sink. Source objects can be specified using single wildcard, as
+well as based on the file modification date.
+
+The way this operator works can be compared to the ``cp`` command.
+
+GoogleCloudStorageSynchronizeBuckets
+------------------------------------
+
+The :class:`~airflow.operators.gcs_to_gcs.GoogleCloudStorageSynchronizeBuckets`
+operator checks the initial state of the destination bucket, and then compares it with the source bucket.
+Based on this, it creates an operation plan that describes which objects should be deleted from
+the destination bucket, which should be overwritten, and which should be copied.
+
+This operator never deletes data in the destination bucket and it never modifies objects in the source bucket.
```

Comment:
So what the `for blob in to_delete_blobs` does? Is it not deleting files in destination bucket?

Reply:
Good point. I was thinking about a source bucket.